Assets/HoloToolkit/Input/Scripts/Utilities/MotionControllerVisualizer.cs

```diff
@@ -157,23 +157,33 @@ private void UpdateControllerState()
                     }
 
                     Vector3 newPosition;
-                    if (sourceState.sourcePose.TryGetPosition(out newPosition, InteractionSourceNode.Grip))
+                    if (sourceState.sourcePose.TryGetPosition(out newPosition, InteractionSourceNode.Grip) && ValidPosition(newPosition))
                     {
                         currentController.ControllerParent.transform.localPosition = newPosition;
                     }
 
                     Quaternion newRotation;
-                    if (sourceState.sourcePose.TryGetRotation(out newRotation, InteractionSourceNode.Grip))
+                    if (sourceState.sourcePose.TryGetRotation(out newRotation, InteractionSourceNode.Grip) && ValidRotation(newRotation))
                     {
                         currentController.ControllerParent.transform.localRotation = newRotation;
                     }
                 }
             }
 #endif
         }
-
+		private bool ValidRotation(Quaternion newRotation)
```

Comment:
Please double check tabs -> spaces :)

Reply:
I am using the default VS 2017 formatting. What should I use?

Reply:
Tab size is 4

Reply:
Yep, the Toolkit uses 4 spaces as a tab. Several of the added lines in here (such as the one this comment is on) are actual tabs though.

Reply:
Found it. Tabify is on.